src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -451,6 +451,14 @@ private RecommendedSymbols GetSymbolsOffOfConditionalReceiver(ExpressionSyntax o
                 ? symbols.WhereAsArray(s => !(s.IsStatic || s is ITypeSymbol))
                 : (abstractsOnly ? symbols.WhereAsArray(s => s.IsAbstract) : symbols);
 
+
+            //If container type is "ref struct" then we should exclude methods from object and ValueType that are not overriden,
+            //because calling them produces a compiler error due to unallowed boxing. See https://github.com/dotnet/roslyn/issues/35178
+            if (containerType is not null && containerType.IsRefLikeType)
+            {
+                namedSymbols = namedSymbols.RemoveAll(s => s.ContainingType.SpecialType is SpecialType.System_Object or SpecialType.System_ValueType);
+            }
```

Comment:
i'm not sure why this is on the IDE side.  the compiler is including these members so it seems like the fix should be there if they're not acutally valid to be used.  RIght?  @RikkiGibson what do you think?

Reply:
This is the only place, where the type symbol for which completions are requested is obtained. If to look at the call stack, the first place of `Features` layer, which is called, is this: https://github.com/dotnet/roslyn/blob/accb30658c0d3dbaa9e16929816074a496a5ed2a/src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs#L37 Here recommended symbols are asked for, but after obtaining them there is no way to know, for what type symbol they are

Reply:
the symbols were presumably returned by `_context.SemanticModel.LookupSymbols(position, container, includeReducedExtensionMethods: true)`.  Right?

Reply:
Maybe. But looking at current code structure, `Features` layer only asks `Workspaces` layer to return recomendations for the given document and position, while `Workspaces` layer gets syntax expression, containig named symbol and returns it's content. Furthermore, there was some logic applied to returning recomendations even before this PR. For instance, the main focus of the issue I fixed is `Span<T>` recomendations. I noticed that even before my change there was no "GetHashCode" recomendation, even though it is one of the `object`'s method. It turned out, that it was excluded because of being marked as obsolete, and this logic was applied in `Workcpaces` layer. _Maybe_ this system needs a bit of redisign, but if I extract my peace of logic to `Features` layer it will get worse from the architectural point of view as the logic will get spread across several layers

Reply:
@DoctorKrolic this feels like a potential issue with the core compiler API here to me.

>  Maybe. But looking at current code structure, Features layer only asks Workspaces layer to return recomendations for the given document 

And the workspaces layer asks the compiler layer for the available members at that point. 

> but if I extract my peace of logic to Features layer it will get worse from the architectural point of view as the logic will get spread across several layers

I'm not asking for it to go in the features layer. I think it may need to go in the compiler layer. 

>  will get worse from the architectural point of view as the logic will get spread across several layers

Code should exist at the relevant layer. If it is not legal for this member to be referenced by the language, then I think it would be sensible for the compiler to not return it. Checking with @RikkiGibson for thoughts here.

Reply:
Technically you can't invoke `object.ToString()`, `ValueType.ToString()` and `Span<T>.ToString()` separately, you can only invoke `ToString()` and what implementation will be used is determined during compilation or runtime. But compiler layer still returns you all possible implementations, even when it is obvious that some of them are not valid for this case. Then overrides are filtered in `Workspace` layer. Here is raw member list for empty struct: ![devenv_LYiIbnEsN3](https://user-images.githubusercontent.com/70431552/156619218-a99d6959-d5e5-436b-89f4-ce9c8d9c39a5.png)
I got your idea, but still not sure if it is worth doing or not. Waiting for the third opinion now as well

Reply:
I think it's reasonable to exclude `ValueType` and `Object` members on ref structs in the `LookupSymbols` API just like we exclude inaccessible members.

Here the issue is that we can't convert the type to its base type. I'm not sure there's another precise analogue to base our decision on--"member is accessible, yet it can never be used on this container". But it does seem like we should limit the results to "symbols that could *possibly* be used on this receiver in this context".

Reply:
Done

Reply:
As an example, please add this as a test and make sure that `ToString` appears in the completion list:

```cs
C c = new C();
_ = nameof(c.$$);

public ref struct C
{
}
```

Reply:
Perhaps it would make sense to do the filtering in the same area that e.g. EditorBrowsable attributes are handled. Was that the original area the PR was changing?